python_modules/dagster/dagster/_core/definitions/metadata/metadata_value.py

```diff
@@ -1017,3 +1030,29 @@ class NullMetadataValue(NamedTuple("_NullMetadataValue", []), MetadataValue[None
     def value(self) -> None:
         """None: The wrapped null value."""
         return None
+
+
+@whitelist_for_serdes
+class CodeLocationStateMetadataValue(
```

Comment:
`ForDefinitionsReconstructionMetadataValue`? I think that might be best.

Reply:
For the record we decided on `CodeLocationReconstructionMetadataValue`